chain/actors/builtin/miner/actor.go.template

```diff
@@ -102,12 +103,21 @@ type State interface {
 	DeadlineInfo(epoch abi.ChainEpoch) (*dline.Info, error)
 	DeadlineCronActive() (bool, error)
 
-	// Diff helpers. Used by Diff* functions internally.
-	sectors() (adt.Array, error)
-	decodeSectorOnChainInfo(*cbg.Deferred) (SectorOnChainInfo, error)
-	precommits() (adt.Map, error)
-	decodeSectorPreCommitOnChainInfo(*cbg.Deferred) (miner{{.latestVersion}}.SectorPreCommitOnChainInfo, error)
 	GetState() interface{}
+
+	SectorsArray() (adt.Array, error)
+	SectorsAmtBitWidth() int
+	DecodeSectorOnChainInfo(*cbg.Deferred) (SectorOnChainInfo, error)
```

Comment:
Discussed offline I want all the things that need a decode method like this to just become top level interface types (like Deadline Partition below) that have the needed accessor interface to meet diffing needs.

Reply:
@magik6k do you think this is necessary?  How do you feel about merging without changing from this public Decode method?

Reply:
I don't love this, but I see why we need this; Ideally we'd have a better/more consistent way to go through state at any version.

Reply:
Would you be willing to punt addressing this to a different PR? 😁 🙏 
 I don't love it either, and wrapping it with an interface as Zen suggested would certainly be better, but the work to do so is nontrivial.

Reply:
I...would prefer not to punt this because I'm worried it'll never happen, and this is fairly ugly. I could help with the wrangling if that would make it easier.

Reply:
I'd love help wrangling this as I am not sure what shape it should be in, would you mind doing that bit?

Reply:
@frrist lotus team wont have bandwidth to do this until we ship v1.18.0 - so if this is something you need in this release, I’m afraid you have to take this on yourself 🙏